protocol/x/clob/e2e/order_matches_test.go

```diff
@@ -0,0 +1,90 @@
+package clob_test
+
+import (
+	"testing"
+
+	"github.com/cometbft/cometbft/types"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	clobtestutils "github.com/dydxprotocol/v4-chain/protocol/testutil/clob"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	feetiertypes "github.com/dydxprotocol/v4-chain/protocol/x/feetiers/types"
+	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+)
+
+func TestOrderMatches(t *testing.T) {
+	tests := map[string]struct {
+		subaccounts       []satypes.Subaccount
+		blockAdvancements []testapp.BlockAdvancementWithError
+	}{
+		"Error: partially filled IOC taker order cannot be matched twice": {
```

Comment:
Suggestion: can we add the following two tests?

- Order is partially-filled, then replaced with an FOK order and the FOK order is rejected.
- IOC order is filled as a taker across multiple matches.